modules/neovim/init/basic.nix

```diff
@@ -158,112 +159,159 @@ in {
       default = "sensitive";
       description = "Set the case sensitivity of search";
     };
+
+    undoFile = {
+      enable = mkEnableOption ''
+        undofile for Neovim.
+
+        ::: {.warning}
+        [{option}`vim.undoFile.path`](#opt-vim.undoFile.path) **must** be set to a valid
+        path for this option to have any effect.
+        :::
+      '';
+
+      path = mkOption {
+        type = nullOr (either str luaInline);
+        default = null;
+        example = "os.getenv('XDG_DATA_HOME') .. '/nvf/undo'";
```

Comment:
We should use `vim.fn.stdpath`, specifically `vim.fn.stdpath("state")`, which should be fine to have as a default, not as an example. See `:help stdpath` for more information.

Also, you forgot to pass this string to `mkLuaInline`.